datasets/told-br/told-br.py

```diff
@@ -0,0 +1,178 @@
+ToLD-Br is the biggest dataset for toxic tweets in Brazilian Portuguese, crowdsourced
+by 42 annotators selected from a pool of 129 volunteers. Annotators were selected aiming
+to create a plural group in terms of demographics (ethnicity, sexual orientation, age, gender).
+Each tweet was labeled by three annotators in 6 possible categories:
+LGBTQ+phobia,Xenophobia, Obscene, Insult, Misogyny and Racism.
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = "https://github.com/JAugusto97/ToLD-Br"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "https://github.com/JAugusto97/ToLD-Br/blob/main/LICENSE_ToLD-Br.txt "
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace Datasets library doesn't host the datasets but only points to the original files.
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+
+_URLS = {
+    "multilabel": "https://raw.githubusercontent.com/JAugusto97/ToLD-Br/main/ToLD-BR.csv",
+    "binary": "https://github.com/JAugusto97/ToLD-Br/raw/main/experiments/data/1annotator.zip",
+}
+
+
+class ToldBr(datasets.GeneratorBasedBuilder):
+    """Toxic/Abusive Tweets Classification Dataset for Brazilian Portuguese."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="multilabel",
+            version=VERSION,
+            description="""
+            Full multilabel dataset with target values ranging
+            from 0 to 3 representing the votes from each annotator.
+            """,
+        ),
+        datasets.BuilderConfig(
+            name="binary",
+            version=VERSION,
+            description="""
+            Binary classification dataset version separated in train, dev and test test.
+            A text is considered toxic if at least one of the multilabel classes were labeled
+            by at least one annotator.
+            """,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "binary"
+
+    def _info(self):
+        if self.config.name == "binary":
+            features = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "toxic": datasets.Value("uint32"),
```

Comment:
Could you use a ClassLabel type here for consistency with the other classification datasets ?
```suggestion
                    "label": datasets.ClassLabel(names=["not-toxic", "toxic"]),
```

See the [ethos](https://huggingface.co/datasets/ethos/blob/main/ethos.py) dataset for example